auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -726,8 +784,10 @@ private Void replaceIdentityPrivileged(final LoadedIdentity newIdentity) throws
                     try (OutputStream outputStream = new BufferedOutputStream(Files.newOutputStream(tempPath, WRITE, CREATE_NEW, DSYNC))) {
                         try (AutoCloseableXMLStreamWriterHolder holder = new AutoCloseableXMLStreamWriterHolder(xmlOutputFactory.createXMLStreamWriter(outputStream))) {
                             writeIdentity(holder.getXmlStreamWriter(), newIdentity);
-                        } catch (XMLStreamException | InvalidKeySpecException | NoSuchAlgorithmException | CertificateEncodingException e) {
+                        } catch (XMLStreamException e) {
```

Comment:
Just minor, but I think here is similar issue with the exceptions? IIUC the `NoSuchAlgorithmException` and `InvalidKeySpecException` could have been thrown previously in `writeIdentity` method during encoding of the  password for identity. So we are changing the resulting error message for those cases to be` "Filesystem-backed realm unable to encrypt identity."` which could be misguiding. This could also be avoided by surrounding just the password encoding in `writeIdentity` and throwing appropriate exception there.

Reply:
Good catch, @Skyllarr!

I believe we could just add `InvalidKeySpecException | NoSuchAlgorithmException | CertificateEncodingException` back here.

I think the `GeneralSecurityException` that gets thrown in the case of an encryption/decryption error is different than the above three exceptions so we'll still get to the `catch (GeneralSecurityException e)` part right after that. (@Ashpan please test to make sure that's the case.)

Reply:
I think I may have removed these exception catches unintentionally. If I were to add them back, and writeIdentity hits an `InvalidKeySpecException` then it would get caught here and throw a `fileSystemRealmFailedToWrite(...)` instead right? Which would avoid the incorrect error message?

Reply:
Oh just saw your comment @fjuma, will test, and the exceptions back if it works as intended

Reply:
Added back the `InvalidKeySpecException | NoSuchAlgorithmException | CertificateEncodingException` exceptions and it works as intended.